### PR TITLE
Rotosolve warning no trainables

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
As discussed in #2100 , the warning that `RotosolveOptimizer` will only update arguments marked as trainable is made more visible in this PR. Additionally, a warning is raised if no trainable arguments are found.

**Description of the Change:**
Doc string improvement (enhance visibility of existing warning regarding training).
Introduce a warning for use cases in which the optimizer would not do anything.

**Benefits:**
UI improvement

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
#2100

### CODE REVIEW

File: pennylane/optimize/rotosolve.py [hunk 1]
```diff
@@ -237,6 +242,12 @@ def cost_function(rot_param, layer_par, crot_param, rot_weights=None, crot_weigh
     With TensorFlow, a ``tf.Variable`` inside a ``tf.GradientTape`` may be used to
     mark variables as trainable.
 
+    .. warning::
+
+        ``RotosolveOptimizer`` will only update parameters that are *explicitly*
+        marked as trainable. Either via ``requires_grad`` if using Autograd or PyTorch,
+        or by using `tf.Variable` tensors inside a `GradientTape` if using TensorFlow.
```
Comment (josh146) [inline pennylane/optimize/rotosolve.py:249]: Although does this conflict with the warning below that the Rotosolve optimizer doesn't work with TF?

**Review by josh146 — APPROVED** [review]
Thanks @dwierichs, this is a nice usability change!

File: pennylane/optimize/rotosolve.py (hunk 1 above)
Comment (dwierichs) [reply pennylane/optimize/rotosolve.py:249]: I'm not entirely sure I understand the conflict, sorry. I'm seeing the above `..warning::` text and the text above the warning that states to use a GradientTape together with `tf.Variable` to mark parameters as trainable. Am I missing a warning?

Comment (josh146) [reply pennylane/optimize/rotosolve.py:249]: Oh I'm just thinking of this: https://github.com/PennyLaneAI/pennylane/issues/2083

Comment (dwierichs) [reply pennylane/optimize/rotosolve.py]: Very nice catch, I edited the warning and moved it further up in the docstring to hopefully catch all users coming across this incompatibility. :)

**Pull request merged**